modules/pipeline/providers/cms/db/op.go

```diff
@@ -92,6 +92,17 @@ func (client *Client) GetCmsNs(pipelineSource apistructs.PipelineSource, ns stri
 	return cmsNs, exist, nil
 }
 
+func (client *Client) BatchGetCmsNamespaces(pipelineSource []string, ns []string, ops ...mysqlxorm.SessionOption) ([]PipelineCmsNs, error) {
+	session := client.NewSession(ops...)
+	defer session.Close()
+
+	var namespaces []PipelineCmsNs
+	if err := session.In("pipeline_source", pipelineSource).In("ns", ns).Find(&namespaces); err != nil {
```

Comment:
You will get many unexpected result.
I think the size of source can only be one.